src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java

```diff
@@ -9827,4 +9833,25 @@ public static Value zap(ScriptRuntime controller, Value item) {
 
     return DataTypes.makeItemValue(itemId, true);
   }
+
+  public static Value get_autumnaton_locations(ScriptRuntime controller) {
+    if (!InventoryManager.hasItem(ItemPool.AUTUMNATON)) {
+      return new ArrayValue(new AggregateType(DataTypes.LOCATION_TYPE, 0));
```

Comment:
Does KoL only supply the list if the AUTUMNATON is in inventory?  I can see a use case for planning where it might be possessed but elsewhere as well as a case where the locations are of interest even if the item is not owned.  Not a blocker.

Reply:
at the moment, there's no caching, and we need to successfully use the autumn-aton to get the list of locations. I want to add caching, but I need to do some more investigation first.